python-package/xgboost/spark/core.py

```diff
@@ -1027,22 +1049,48 @@ def predict_udf(iterator: Iterator[pd.DataFrame]) -> Iterator[pd.Series]:
                 else:
                     base_margin = None
 
+                data = {}
                 preds = model.predict(
                     X,
                     base_margin=base_margin,
                     validate_features=False,
                     **predict_params,
                 )
-                yield pd.Series(preds)
+                data[pred.prediction] = pd.Series(preds)
+
+                if pred_contrib_col_name:
+                    contribs = pred_contribs(model, X, base_margin)
```

Comment:
Q: Possible optimization: can we compute `pred_contribs`, `proba`, and `prediction` in one pass ?

Reply:
AFAIK, the xgboost API can predict 1 type at one time. @trivialfis can you correct me?

Reply:
That's correct. However, the Spark API can iterate through multiple types of prediction for meeting the spark convention if needed.